DiagnosticsExtension/Models/ConnectionStringValidator/BlobStorageValidator.cs

```diff
@@ -0,0 +1,196 @@
+                    {
+                        response.Status = ConnectionStringValidationResult.ResultStatus.SystemAssignedmanagedidentity;
+                    }
+                }
+                else if (e.Message.Contains("ManagedIdentityCredential"))
+                {
+                    response.Status = ConnectionStringValidationResult.ResultStatus.ManagedIdentityCredential;
+                }
+                else if (e.Message.Contains("fullyQualifiedNamespacemissed"))
+                {
+                    response.Status = ConnectionStringValidationResult.ResultStatus.FullyQualifiedNamespacemissed;
+                }
+                else if (e is EmptyConnectionStringException)
+                {
+                    response.Status = ConnectionStringValidationResult.ResultStatus.EmptyConnectionString;
+                }
+                else if (e.InnerException != null &&
+                         e.InnerException.Message.Contains("The remote name could not be resolved"))
+                {
+                    response.Status = ConnectionStringValidationResult.ResultStatus.DnsLookupFailed;
+                }
+                else if (e is StorageException)
+                {
+                    if (((StorageException)e).RequestInformation.HttpStatusCode == 401)
+                    {
+                        response.Status = ConnectionStringValidationResult.ResultStatus.AuthFailure;
+                    }
+                    else if (((StorageException)e).RequestInformation.HttpStatusCode == 403)
+                    {
+                        response.Status = ConnectionStringValidationResult.ResultStatus.Forbidden;
+                    }
+                }
+                else
+                {
+                    response.Status = ConnectionStringValidationResult.ResultStatus.UnknownError;
+                }
+                response.Exception = e;
+            }
+
+            return response;
+        }
+
+        public async Task<TestConnectionData> TestConnectionStringViaAppSetting(string appSettingName, string entityName)
+        {
+            var envDict = Environment.GetEnvironmentVariables();
+            string appSettingClientIdValue, appSettingClientCredValue = null;
+            string value = null;
+            BlobServiceClient client = null;
+
+            if (envDict.Contains(appSettingName))
+            {
+                try
+                {
+                    value = Environment.GetEnvironmentVariable(appSettingName);
+                    client = new BlobServiceClient(value);
+                }
+                catch (ArgumentNullException e)
+                {
+                    throw new EmptyConnectionStringException(e.Message, e);
+                }
+                catch (Exception e)
+                {
+                    throw new MalformedConnectionStringException(e.Message, e);
+                }
+            }
+            else
+            {
+                try
```

Comment:
Would improve code readability to have comments for each logical block below indicating the special case its handling.  e.g.
// System assigned managed identity detected

// User assigned managed identity detected

Reply:
implemented for blob storage validator, will do for other validators as well.

Reply:
fixed